### PR TITLE
Add templates support

### PR DESCRIPTION
**Pull request by Tapped**

### CODE REVIEW

File: src/css/dracula.css [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+    Name:       dracula
+    Author:     Michael Kaminsky (http://github.com/mkaminsky11)
+
+    Original dracula color scheme by Zeno Rocha (https://github.com/zenorocha/dracula-theme)
+
+*/
+
+
+.cm-s-dracula.CodeMirror, .cm-s-dracula .CodeMirror-gutters {
```
Comment (Tapped) [inline src/css/dracula.css:11]: It's the same css as found in CodeMirror but with the background-color set to same as the input field color in Grafana.

**Comment by polomani** [comment]
what are your intentions here? aren't templates going to be deprecated soon?

**Comment by Tapped** [comment]
> what are your intentions here? aren't templates going to be deprecated soon?

Soon doesn't exist in software service world :) It'll probably take a couple of quarters, so the intention is to continue to help drive use cases through templates, and also help sales.

File: src/datasources/TemplatesDatasource.ts [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+
+  private runQuery(
+    query: TemplateQuery & { refId: string },
+    range: TimeRange | undefined,
+    scopedVars: ScopedVars | undefined = undefined
+  ) {
+    let payload = query.graphQlQuery;
+    if (range) {
+      payload = payload.replace(/\$timeFrom/g, range.from.valueOf().toString());
+      payload = payload.replace(/\$timeTo/g, range.to.valueOf().toString());
```
[the remark this answers is not in the document]
Comment (Tapped) [reply src/datasources/TemplatesDatasource.ts:81]: Good question, can probably be removed.

**Review by muradsater — APPROVED** [review]

**Comment by andeplane** [comment]
> what are your intentions here? aren't templates going to be deprecated soon?

Templates in its current state is what we have available in 2022 with a replacement reaching GA late 2022 at earliest. The replacement will contain very similar features, and we will do everything in our power to help with migration and a smooth transition. We may even replace the templates backend, but keep the APIs until we have migrated all users.

Templates deliver lots of great value for many users today :)

**Comment by Tapped** [comment]
image

@polomani thanks to @muradsater we have added a "tag" with `Preview` to the templates tab. Do you think we are ready to merge and release it?

**Comment by muradsater** [comment]
@polomani Added the opt-in, let me know if this solution is ok

**Review by polomani — APPROVED** [review]
LGTM

@Tapped I moved some things around, could you do a quick test, if there are no regressions?

**Pull request merged**